### ISSUE TITLE
Google Cloud Postgres not Capturing Data

### ISSUE THREAD

**Issue by dyaffe** [estuary/flow#398]
I've been testing Google Cloud Postgres and theoretically we should be able to capture data from there using the steps in the proper section of `[this doc](https://docs.google.com/document/d/18tLqjKmdVSyqGtTgXOdoHXrH_9OsqNumapmP5FVi3SE/edit?usp=sharing)`.

This of course requires a SSH tunnel to connect with the DB.  I've set up a catalog which attempts to capture data using that SSH connector and push it to a local postgres DB.  

[flow.txt](https://github.com/estuary/flow/files/8188095/flow.txt)

Note that this capture is using Phil's [VPN proxy](https://console.cloud.google.com/compute/instancesDetail/zones/us-central1-a/instances/phil-dev?project=helpful-kingdom-273219).  That has an ephemeral IP and will have to be changed in the catalog if it restarts.  I'm also using [this](https://console.cloud.google.com/sql/instances/dave-test-cdc-instance/overview?cloudshell=false&project=helpful-kingdom-273219) specific Google Cloud Postgres DB.  

What actually happens after running this catalog is that it succeeds but no data is captured.  It seems to get some capture failures that look like this:
WARN[0019] read failure (will retry)                     attempt=0 err="rpc error: code = Unavailable desc = all SubConns are in TransientFailure" journal=recovery/capture/acmeCo/source-postgres/00000000-00000000 offset=0
WARN[0019] read failure (will retry)                     attempt=0 err="rpc error: code = Unavailable desc = all SubConns are in TransientFailure" journal=recovery/materialize/acmeCo/postgres/materialize-data/00000000-00000000 offset=0

I'm also seeing an "unhandled global request" in the capture journal logs:
{"_meta":{"uuid":"a7fe1ece-9be5-11ec-8400-49fddbd7be81"},"shard":{"name":"acmeCo/source-postgres","kind":"capture","keyBegin":"00000000","rClockBegin":"00000000"},"ts":"2022-03-04T18:05:20.990753166Z","level":"info","message":"Unhandled global request: Ok(\"hostkeys-00@openssh.com\")","fields":{"log.file":"/home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/thrussh-0.33.5/src/client/encrypted.rs","log.line":481,"log.module_path":"thrussh::client::encrypted","log.target":"thrussh::client::encrypted","logSource":"connector stderr","operation":"read"}}

I'm not 100% sure what's going on here but it may be helpful to have @jixij or @willdonnelly take a quick looks depending on whether this is a proxy problem or a capture one.

**Comment by willdonnelly** [comment]
This looks like _probably_ something related to SSH proxying rather than the capture process itself, simply judging by that last error message which mentions `hostkeys-00@openssh.com` and the Rust `thrussh` crate. I haven't dug into what that particular error message might mean though.

**Comment by jixij** [comment]
`Unhandled global request` error looks benign to me. Checking on the others...

**Comment by jixij** [comment]
I can reproduce the issue locally. I am able to build and activate the catalog, and I did not see any errors generated.

However, I tried to materialize the data to  `postgres` and `sqlite3`, and both gets no data. So seems like it is more related to the setting of the source db.

I logged in to the db, and I found some permission issues when trying to add an addition record to the testing tabe.

```
postgres=> select * from replica_test;
 id |  data  
----+--------
  1 | apple
  2 | banana
  3 | cherry
  4 | apple
  5 | banana
  6 | cherry
(6 rows)

postgres=> insert into replica_test(id, data) values(7, 'test');
ERROR:  permission denied for table replica_test
postgres=> 
```

Does that matter? @willdonnelly

**Comment by dyaffe** [comment]
That doesn't seem like the issue to me since the user we're working with is flow_capture and it seems to have the proper permissions:

postgres=> SELECT grantee, privilege_type 
FROM information_schema.role_table_grants 
WHERE table_name='flow_watermarks';
   grantee    | privilege_type
--------------+----------------
 flow_capture | INSERT
 flow_capture | SELECT
 flow_capture | UPDATE
 flow_capture | DELETE
 flow_capture | TRUNCATE
 flow_capture | REFERENCES
 flow_capture | TRIGGER

The Google Cloud Posgres section of [this doc](https://docs.google.com/document/d/18tLqjKmdVSyqGtTgXOdoHXrH_9OsqNumapmP5FVi3SE/edit) lists the steps that I used to set this up.

**Comment by willdonnelly** [comment]
After debugging this a bit over VC, we figured out that the problem was that the `watermarksTable` config parameter was set to `flow_watermarks` and it needs to be `public.flow_watermarks` for correct operation. I'm going to go file an issue for improving that, just brainstorming I see three possibilities:

1. Log a warning or error if the `watermarksTable` config parameter doesn't contain a period, since we currently expect it to be fully-qualified.
2. Make the "is this a watermark change" logic more permissive. We're only ever looking for one specific UUID that we randomly generated ourselves (or the string "nonexistent-watermark" which could be changed so that it's not so easy to collide with), so in principle it's unlikely to cause trouble even if we checked every single change event for the watermark UUID.
3. Log a warning or error when in `streamToWatermark()` a substantial amount of time passes with no new change events.
